drivers/lora/sx1276.c

```diff
@@ -0,0 +1,717 @@
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static void sx1276_irq_callback(struct device *dev,
+				struct gpio_callback *cb, u32_t pins)
+{
+	struct sx1276_data *dev_data =
+		CONTAINER_OF(cb, struct sx1276_data, irq_cb);
+	int ret;
+	u8_t regval;
+
+	ret = sx1276_read(dev_data->dev, SX1276_REG_IRQ_FLAGS, &regval, 1);
+	if (ret < 0) {
+		LOG_ERR("Unable to read IRQ_FLAGS");
+		return;
+	}
+
+	if (regval & LORA_REG_IRQ_FLAGS_TX_DONE) {
+		LOG_DBG("TX IRQ");
+	}
+
+	if (regval & LORA_REG_IRQ_FLAGS_RX_DONE) {
+		LOG_DBG("RX IRQ");
+		if (regval & 0x20)
+			LOG_DBG("Payload CRC Error");
+
+		if (regval & 0x10)
+			LOG_DBG("Valid header found");
+		else
+			LOG_ERR("No Valid header found");
+	}
+
+	ret = sx1276_write(dev_data->dev, SX1276_REG_IRQ_FLAGS, regval);
+	if (ret < 0) {
+		LOG_ERR("Unable to write IRQ_FLAGS");
+		return;
+	}
+
+	k_sem_give(&dev_data->data_sem);
+}
+
+static int sx1276_lora_init(struct device *dev)
+{
+	struct sx1276_data *data = dev->driver_data;
+	static struct spi_cs_control spi_cs;
+	struct device *reset_dev;
+	int ret;
+	u8_t regval;
+
+	data->spi = device_get_binding(DT_INST_0_SEMTECH_SX1276_BUS_NAME);
+	if (!data->spi) {
+		LOG_ERR("spi device not found: %s",
+			    DT_INST_0_SEMTECH_SX1276_BUS_NAME);
+		return -EINVAL;
+	}
+
+	data->spi_cfg.operation = SPI_WORD_SET(8) | SPI_TRANSFER_MSB;
+	data->spi_cfg.frequency = DT_INST_0_SEMTECH_SX1276_SPI_MAX_FREQUENCY;
+	data->spi_cfg.slave = DT_INST_0_SEMTECH_SX1276_BASE_ADDRESS;
+
+	spi_cs.gpio_pin = GPIO_CS_PIN,
+	spi_cs.gpio_dev = device_get_binding(
+			DT_INST_0_SEMTECH_SX1276_CS_GPIOS_CONTROLLER);
+	if (!spi_cs.gpio_dev) {
+		LOG_ERR("Failed to initialize CS GPIO driver: %s",
+		       DT_INST_0_SEMTECH_SX1276_CS_GPIOS_CONTROLLER);
+		return -EIO;
+	}
+
+	data->spi_cfg.cs = &spi_cs;
+
+	/* Setup DIO gpio */
+	data->dio0 = device_get_binding(GPIO_DIO0_CONTROLLER);
+	if (data->dio0 == NULL) {
+		LOG_ERR("Cannot get pointer to %s device",
+			GPIO_DIO0_CONTROLLER);
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(data->dio0, GPIO_DIO0_PIN,
+			   GPIO_DIR_IN | GPIO_INT | GPIO_INT_EDGE |
+			   GPIO_INT_DEBOUNCE | GPIO_INT_ACTIVE_HIGH);
+
+	gpio_init_callback(&data->irq_cb,
+			   sx1276_irq_callback,
+			   BIT(GPIO_DIO0_PIN));
+
+	if (gpio_add_callback(data->dio0, &data->irq_cb) < 0) {
+		LOG_ERR("Could not set gpio callback.");
+		return -EIO;
+	}
+	gpio_pin_enable_callback(data->dio0, GPIO_DIO0_PIN);
+
+	/* Setup Reset gpio */
+	reset_dev = device_get_binding(
+			DT_INST_0_SEMTECH_SX1276_RESET_GPIOS_CONTROLLER);
+	if (!reset_dev) {
+		LOG_ERR("Failed to initialize Reset GPIO driver: %s",
+		       DT_INST_0_SEMTECH_SX1276_RESET_GPIOS_CONTROLLER);
+		return -EIO;
+	}
+
+	ret = gpio_pin_configure(reset_dev, GPIO_RESET_PIN, GPIO_DIR_OUT);
+
+	/* Perform soft reset */
+	gpio_pin_write(reset_dev, GPIO_RESET_PIN, 0);
+	k_sleep(100);
+	gpio_pin_write(reset_dev, GPIO_RESET_PIN, 1);
+	k_sleep(100);
+
+	ret = sx1276_read(dev, SX1276_REG_VERSION, &regval, 1);
```

Comment:
I have a custom board using the STM32L151CBT.
Here, the `regval` value is not initialized, so my case is `0xE0`.
VERSION is read correctly, but `MOSI` is outputting `0xE0`.
So the waveform appears as below.

**[u8_t regval;]**
![scope_30](https://user-images.githubusercontent.com/10510127/62431368-016b5480-b762-11e9-8f7d-462b41e066d5.png)


**[u8_t regval = 0x00;]**
![scope_31](https://user-images.githubusercontent.com/10510127/62431370-08926280-b762-11e9-8f12-6a0cfdf43d8d.png)